prow/cip/config.yaml

```diff
@@ -0,0 +1,153 @@
+# Basically anything with a "name: ..." and "spec: ..." is a Prow Job. Prow
```

Comment:
IMO this should go in something like `config/cip/`, our specific configurations are slowly making their way to a heirarchy in `config/`, decoupled from the underlying tools. eventually prow will probably move to it's own repo.

Reply:
Ack. I guess the main question I have is --- can we create a new Prow cluster that uses these configs today, or should we piggyback into the existing one https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml ? From your comment I'm going to assume the former (new Prow cluster)

Reply:
the config you have here is for a full prow cluster. the presets and jobs could be moved into configs in config/jobs/ and use the existing prow, but we'd need to add webhooks for your repo at minimum.

Reply:
Ah, makes sense. I'll go ahead and break it up into config/jobs. :sweat_smile: 

As for the plugins.yaml in this PR, I would need to move it into the [main one](https://github.com/kubernetes/test-infra/blob/master/prow/plugins.yaml) though, right?